src/column_family.rs

```diff
@@ -43,6 +43,7 @@ impl ColumnFamilyDescriptor {
 
 /// An opaque type used to represent a column family. Returned from some functions, and used
 /// in others
+#[derive(Clone)]
```

Comment:
Do you really think it is safe ? 🤔

Reply:
I believe so, here's my analysis:

```
pub struct ColumnFamily {
    pub(crate) inner: *mut ffi::rocksdb_column_family_handle_t,
}
```

where inner is a rocksdb pointer to a C++ object: https://github.com/facebook/rocksdb/blob/master/db/c.cc#L973-L974 `rocksdb_column_family_handle_t* handle = new rocksdb_column_family_handle_t;`, which aliases an object defined here: 

```
class ColumnFamilyHandle {
 public:
  virtual ~ColumnFamilyHandle() {}
  // Returns the name of the column family associated with the current handle.
  virtual const std::string& GetName() const = 0;
  // Returns the ID of the column family associated with the current handle.
  virtual uint32_t GetID() const = 0;
  // Fills "*desc" with the up-to-date descriptor of the column family
  // associated with this handle. Since it fills "*desc" with the up-to-date
  // information, this call might internally lock and release DB mutex to
  // access the up-to-date CF options.  In addition, all the pointer-typed
  // options cannot be referenced any longer than the original options exist.
  //
  // Note that this function is not supported in RocksDBLite.
  virtual Status GetDescriptor(ColumnFamilyDescriptor* desc) = 0;
  // Returns the comparator of the column family associated with the
  // current handle.
  virtual const Comparator* GetComparator() const = 0;
};
```

where the destructor  `~ColumnFamilyHandle()` is what decrements the refcount to the column family and performs cleanup of old files: https://github.com/facebook/rocksdb/blob/master/db/column_family.cc#L68-L84

The destructor however is invoked through `rocksdb_column_family_handle_destroy` https://github.com/facebook/rocksdb/blob/master/db/c.cc#L997-L1000, which is not what's called in the wrapper's `drop_cf`: https://github.com/rust-rocksdb/rust-rocksdb/blob/master/src/db.rs#L687, which is only called when the `DB` object is dropped: 

```
fn drop(&mut self) {
        unsafe {
            self.cfs.get(|cf_map| {
                for cf in cf_map.values() {
                    ffi::rocksdb_column_family_handle_destroy(cf.inner);
                }
                ffi::rocksdb_close(self.inner);
            })
        }
    }
```

This means cloning the handle here should be a harmless operation that does not incur any cleanup or additional data races.

Reply:
@aleksuss Taking your concern into account, I've removed the `Clone` on the object and instead implemented a separate `cf_handle_multi_threaded()` path that returns an `Arc<ColumnFamily>`. This means only users of the multithreaded config will risk running into the inconsistency you described above, while by default, existing users will be unaffected.